src/analyses/dependence_graph.h

```diff
@@ -242,6 +242,10 @@ class dependence_grapht:
   {
     ait<dep_graph_domaint>::initialize(function, goto_program);
 
+    // The dependency graph requires that all nodes are explicitly created
+    forall_goto_program_instructions(i_it, goto_program)
+      get_state(i_it).make_bottom();
```

Comment:
I need the iterator (i.e. the `locationt`) rather than the instruction themselves.  Is there a version of the ranged-for construct that supports something like `for(goto_programt::targett i_it : goto_program.instructions)`?